src/shared/lending/FortressLendingCore.sol

```diff
@@ -0,0 +1,928 @@
+        ERC4626(_asset, _name, _symbol) {
+
+        (address _collateral, address _oracleMultiply, address _oracleDivide, uint256 _oracleNormalization, address _rateContract,)
+            = abi.decode(_configData, (address, address, address, uint256, address, bytes));
+
+        // Pair Settings
+        assetContract = IERC20(address(_asset));
+        collateralContract = IERC20(_collateral);
+        currentRateInfo.feeToProtocolRate = DEFAULT_PROTOCOL_FEE;
+        cleanLiquidationFee = _liquidationFee;
+        dirtyLiquidationFee = (_liquidationFee * 90000) / LIQ_PRECISION; // 90% of clean fee
+
+        // LTV Settings
+        maxLTV = _maxLTV;
+
+        // Oracle Settings
+        oracleMultiply = _oracleMultiply;
+        oracleDivide = _oracleDivide;
+        oracleNormalization = _oracleNormalization;
+
+        // Rate Calculator Settings
+        rateContract = IRateCalculator(_rateContract);
+
+        // Set swap
+        swap = _swap;
+
+        // Set admins
+        owner = _owner;
+    }
+
+    /// @notice Called immediately after deployment
+    /// @dev This function can only be called by the owner
+    /// @param _rateInitCallData The configuration data for the Rate Calculator contract
+    function initialize(bytes calldata _rateInitCallData) external onlyOwner {
+        // Reverts if init data is not valid
+        IRateCalculator(rateContract).requireValidInitData(_rateInitCallData);
+
+        // Set rate init Data
+        rateInitCallData = _rateInitCallData;
+
+        // Instantiate Interest
+        _addInterest();
+
+        // Instantiate Exchange Rate
+        _updateExchangeRate();
+    }
+
+    // ============================================================================================
+    // External Helpers
+    // ============================================================================================
+
+    /// @notice Returns the total amount of assets managed by Vault (including lent assets)
+    function totalAssets() public view override returns (uint256) {
+        return totalAUM;
+    }
+
+    /// @notice Calculates the shares value in relationship to `amount` and `total`
+    /// @dev Given an amount, return the appropriate number of shares
+    /// @param _totalAmount The total amount of assets
+    /// @param _totalSupply The total supply of shares
+    /// @param _amount The amount of assets to convert to shares
+    /// @param _roundUp Whether to round up or down
+    /// @return _shares The number of shares
+    function convertToShares(uint256 _totalAmount, uint256 _totalSupply, uint256 _amount, bool _roundUp) public pure returns (uint256 _shares) {
+        if (_totalAmount == 0) {
+            _shares = _amount;
+        } else {
+            _shares = (_amount * _totalSupply) / _totalAmount;
+            if (_roundUp && (_shares * _totalAmount) / _totalSupply < _amount) {
+                _shares = _shares + 1;
+            }
+        }
+    }
```

Comment:
Maybe a general remark. How exploitable is the share conversion given arbitrary or wonky input parameters? For instance, what assumptions does the platform make or what assumptions of the platform will break, if e.g. `_amount` is 0 or max uint256? The current test suite does not appear to verify certain "weird" behaviour.

Reply:
We use Foundry Fuzz testing to simulate unexpected input, as discussed.